packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -136,32 +138,54 @@ export class CacheDataClient implements IDataClient {
       `Creating cache client using endpoint: '${this.credentialProvider.getCacheEndpoint()}'`
     );
 
+    const channelConfig: ChannelConfiguration =
+      this.configuration.getChannelConfiguration();
+
+    const channelOptions: Record<string, number> = {
+      'grpc-node.max_session_memory': channelConfig.maxSessionMemoryMB,
+      'grpc.use_local_subchannel_pool': channelConfig.useLocalSubchannelPool
+        ? 1
+        : 0,
+    };
+
+    // Apply keepalive settings only if useKeepAlive is true
+    if (channelConfig.useKeepAlive) {
+      // Assert that keepalive configurations are not undefined
+      if (
+        channelConfig.keepAlivePermitWithoutCalls === undefined ||
+        channelConfig.keepAliveTimeoutMs === undefined ||
+        channelConfig.keepAliveTimeMs === undefined
+      ) {
+        throw new Error(
+          'Keepalive settings must be defined when useKeepAlive is enabled.'
+        );
+      }
+
+      // Default values for keepalive options can be set here, if desired
+      channelOptions['grpc.keepalive_permit_without_calls'] =
+        channelConfig.keepAlivePermitWithoutCalls;
+      channelOptions['grpc.keepalive_timeout_ms'] =
+        channelConfig.keepAliveTimeoutMs;
+      channelOptions['grpc.keepalive_time_ms'] = channelConfig.keepAliveTimeMs;
+    }
+
+    let reinitializeClientEveryInterval = false;
```

Comment:
we can move this to channel config too to avoid the type check below

Reply:
yeah let's just keep this out of this PR for now I think?